src/Application/Messenger/AbstractStateChanged.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace MatchBot\Application\Messenger;
+
+use MatchBot\Application\Assertion;
+
+/**
+ * For now, this is named with the idea that `SalesforceWriteProxyRepository` may soon use it for both donations
+ * and e.g. recurring donation mandates. If we decide against that we could rename it to `AbstractDonationChanged`.
+ */
+abstract class AbstractStateChanged
+{
+    /**
+     * @psalm-suppress PossiblyUnusedProperty Plan is to use `$json` with mandates
+     */
+    public array $json;
```

Comment:
$json is not a very meaningful name. How about `snapshot`? It's a snapshot of the state of the entity at the time the message was generated.